app/src/modules/settings/routes/data-model/field-detail/components/interface.vue

```diff
@@ -117,7 +117,34 @@ export default defineComponent({
 		});
 
 		const selectedInterface = computed(() => {
-			return interfaces.value.find((inter: InterfaceConfig) => inter.id === state.fieldData.meta?.interface);
+			// get the default interface
+			const customInterface: InterfaceConfig | undefined = interfaces.value.find(
+				(inter: InterfaceConfig) => inter.id === state.fieldData.meta.interface
+			);
+
+			if (customInterface) {
+				if (customInterface.component.name === 'interface-dropdown') {
+					// update the interface-value to the specific one to only allow values of the correct type
+					// NOTE: the solution via a deep clone is not ideal,
+					// but the only safe option that came to my mind
+					const deepClone = JSON.parse(JSON.stringify(customInterface)); // deep-copy to bo able to modify safely
+
+					const choicesIndex: number = deepClone.options.findIndex((option: any) => {
+						return option.field === 'choices';
+					});
+					const fieldValueIndex: number = deepClone.options[choicesIndex].meta.options.fields.findIndex(
+						(field: any) => {
+							return field.field === 'value';
+						}
+					);
+
+					deepClone.options[choicesIndex].meta.options.fields[fieldValueIndex].type = state.fieldData.type;
+					return deepClone;
+				}
```

Comment:
We shouldn't override the interface's options here, feels a little hacky. It might actually work just fine if we leave it as a text input for the options of select-dropdown, as all values that can be used can be represented in a text format

Reply:
let's say if it's a timestamp field, and we set value to a random string like `abcd`, it would error when they are creating the item right? Just want to clarify this train of thought even though I don't have an actual solution now

Reply:
> let's say if it's a timestamp field, and we set value to a random string like abcd, it would error when they are creating the item right? Just want to clarify this train of thought even though I don't have an actual solution now

It would, but if the admin configures to have "abc" as the value for a datetime dropdown, that's user-error to me

Reply:
>  that's user-error to me
totally agree. So I should go ahead with the computed value I shared in the conversation above this one?